orion/handlers/http/http.go

```diff
@@ -20,8 +20,30 @@ import (
 	"github.com/golang/protobuf/proto"
 	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
 	opentracing "github.com/opentracing/opentracing-go"
+	"google.golang.org/grpc/metadata"
 )
 
+// grpcMetadataCarrier satisfies both opentracing.TextMapWriter and opentracing.TextMapReader.
+type grpcMetadataCarrier metadata.MD
+
+// ForeachKey conforms to the opentracing.TextMapReader interface.
+func (mc grpcMetadataCarrier) ForeachKey(handler func(string, string) error) (err error) {
+	for key, values := range mc {
+		for _, value := range values {
+			if err = handler(key, value); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// Set conforms to the opentracing.TextMapWriter interface.
```

Comment:
add a comment to explain that this carrier ensures metadata keys are lowercased to conform to the HTTP/2 spec

Reply:
added